models/step_rate_generator.h

```diff
@@ -0,0 +1,261 @@
+/*
+ *  step_rate_generator.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#ifndef STEP_RATE_GENERATOR_H
+#define STEP_RATE_GENERATOR_H
+
+// C++ includes:
+#include <vector>
+
+// Includes from nestkernel:
+#include "connection.h"
+#include "device_node.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "stimulating_device.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+/** @BeginDocumentation
+Name: step_rate_generator - provides a piecewise constant input rate
+
+Description:
+
+The rate_generator provides a piecewise constant rate input to the
+connected rate unit(s). Please note that this input is handled in the same
+way as input from any other rate unit, i.e. it is processed by the input
+function of the receiving rate unit. The amplitude of the rate is changed
+at the specified times. The unit of the rate is Hz.
+
+Parameters:
+
+The following parameters can be set in the status dictionary:
+amplitude_times   list of doubles - Times at which current changes in ms
+amplitude_values  list of doubles - Amplitudes of step current current in
+                                    pA
```

Comment:
The unit of the "rates" is most likely not pA, nor is "current current" a good explanation for them